### PR TITLE
Variables: Add 'jsonwithoutquote' formatting options for variables, and format of variable supports pipeline.

### PR DESCRIPTION
**Pull request by MicroOps-cn**
#51685 
The format of variable supports pipeline. example:
`${keywords:doublequote|percentencode}` like `percentencode(doublequote(keywords))`

The 'jsonwithoutquote' formatting options, example:
```
keyworlds = 'test"1'
String to interpolate: '${keyworlds:json}'
Interpolation result: 'test1\\"'
```
Unlike JSON, the backquotes at both ends are removed.

Sometimes, the URL parameter of link is a JSON, such as ` {"SQL": ["{} |= \"${keyworld}\" "]} `. I want to insert my string variables into this JSON. Quotes and other things need to be escaped using JSON methods, but if you use JSON to format, there will be quotation marks around the obtained string, such as `'"test\\"1"'`. After insertion, it will lead to JSON format confusion. So I suggest introducing this `jsonwithoutquote`

### CODE REVIEW

**Comment by mellieA** [comment]
@MicroOps-cn thanks for your suggestion, I've forwarded it on to the squad for comments

File: public/app/features/templating/formatRegistry.ts [hunk 1]
```diff
@@ -254,6 +255,14 @@ export const formatRegistry = new Registry<FormatRegistryItem>(() => {
         return formatQueryParameter(name, value);
       },
     },
+    {
+      id: FormatRegistryID.jsonWithoutQuote,
+      name: 'JSON',
+      description: 'JSON stringify valu',
```
Comment (JoaoSilvaGrafana) [inline public/app/features/templating/formatRegistry.ts:261]: We should change the text here to explain what it is for.

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Hey @MicroOps-cn 👋 Thanks for submitting this PR and for including tests 😄 It looks good to me, just a small comment about changing the some text!

**Comment by JoaoSilvaGrafana** [comment]
Also please check the [Formatting guidelines](https://github.com/grafana/grafana/blob/main/contribute/create-pull-request.md#formatting-guidelines) for PRs, you just need to add an area before the title of the PR. Maybe something like `Variables: Add 'jsonwithoutquote' formatting option, and format supports pipeline`

File: public/app/features/templating/formatRegistry.ts (hunk 1 above)
Comment (MicroOps-cn) [reply public/app/features/templating/formatRegistry.ts]: It has been modified,

**Review by JoaoSilvaGrafana — APPROVED** [review]
Nice! 🎉

**Pull request merged**